### ISSUE TITLE
i18n strings in artifacts from -G throw errors when run through -A

### ISSUE THREAD

**Issue by connorjclark** [GoogleChrome/lighthouse#9269]
#### Provide the steps to reproduce

`node lighthouse-cli https://www.example.com -G` and when Chrome pops up immediately move it out of focus to force `NO_FCP`

`node lighthouse-cli https://www.example.com -A --view`

#### What is the current behavior?

```
Runtime error encountered: Cannot read property 'icuMessage' of undefined
TypeError: Cannot read property 'icuMessage' of undefined
    at _resolveIcuMessageInstanceId (/Users/cjamcl/src/lighthouse/lighthouse-core/lib/i18n/i18n.js:300:24)
    at replaceInObject (/Users/cjamcl/src/lighthouse/lighthouse-core/lib/i18n/i18n.js:327:55)
    at replaceInObject (/Users/cjamcl/src/lighthouse/lighthouse-core/lib/i18n/i18n.js:340:9)
    at Object.replaceIcuMessageInstanceIds (/Users/cjamcl/src/lighthouse/lighthouse-core/lib/i18n/i18n.js:347:3)
    at Function.run (/Users/cjamcl/src/lighthouse/lighthouse-core/runner.js:155:39)
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:745:11)
    at startup (internal/bootstrap/node.js:282:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:743:3)
```

#### What is the expected behavior?

no error.

**Comment by brendankenny** [comment]
> **ICU replacement errors for -G w/ NO_FCP**
> `Runtime error encountered: Cannot read property 'icuMessage' of undefined`

unfortunately not fixed by #9397. This string is in `LighthouseRunWarnings` in the artifacts. At the end of then running it with `-A`, the localization step tries to look up the string but it's not in `_icuMessageInstanceMap` because the `str()` called happened in the gather phase, not the current run.

However, there's a deeper problem with fixing: even if we did a lookup based on the id, for strings with replacement like the `NO_FCP` warning (`badTraceRecording`), the replacement values are long gone, so the string can't be reconstructed anyways.

We could
- do a replacer/reviver for i18n strings in artifacts
- or some kind of `icuMessagePaths` but for artifacts and swap upon loading them
- or come up with a new `icuMessage` format

we can call this medium priority. This is the only instance I know of after #9397 handles `LHError`s, and only happens if you run *just* `-G` and then *just* `-A` (so basically just us), but it could happen more if we want to e.g. pipe localized strings from gatherers all the way to audits.

**Comment by brendankenny** [comment]
fixed in #10630
